products/platform/documentation-site/research/search-design-doc.md

```diff
@@ -1,15 +1,156 @@
+
+#### Debugging the scraper
+
+#### Debugging the search functionality
+
+We could search for the same term in both the Algolia UI and the landing page to confirm if they both return the same results.
+
+### Caveats
+
+- Access Control: By making the documentation landing page public, it becomes difficult, if not impossible, to allow an unauthenticated user to search documentation from private sources (like a private Github repo). One suggestion is that search results lead to public documentation sources, and we include links to private documentation in the public sources. That way, we can rely on existing access control solutions, while still providing a decent solution. One of the major downsides of that approach is auditing, adding, updating, and maintaining links to/from the public and private sources.
+
+### Security Concerns
+
+If the search input dispatched a request directly to the Algolia API, then there'd be minimal risk of the documentation search negatively affecting other systems. If the request was proxied through our API, then we should be able to rely on any existing DOS protections.
+
+### Privacy Concerns
+
+While there shouldn't be any PII in our public documentation sources, it is possible it exists, and hasn't been discovered yet. Therefore, it is possible that our search results could lead public users to sources that contain PII. If PII is discovered in a repo, we have a [checklist](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/removing-sensitive-data-from-a-repository.md#checklist-for-removing-pii-in-md-file-from-a-documentation-repo) for how to remove it.
+
+### Open Questions and Risks
+
+- Since the documentation landing page will be public only, we will not be able to search private sources of documentation. If searching private documentation becomes a necessary feature in the future, the proposed implementation will likely not be sufficient.
+- How often should we reindex our documentation sources? For perspective, Algolia's open source crawler runs every 24 hours.
+- What should the URL of the documentation landing page be? Since we have `design.va.gov`, does something like `docs.va.gov` make sense?
+- What is the current opportunity cost of building a custom solution? What are we choosing to delay right now by building a custom solution?
+- What is the future opportunity cost of maintaining a custom solution? What will we have to delay in the future by maintaining a custom solution?
+- What is the key person risk? How disruptive would it be if the author(s) of the custom solution were temporarily/permanently unavailable?
+- What is the Not-Invented-Here risk? How likely is that we're reinventing the wheel?
+- What is the inexperience risk? Have any of the team members involved ever built/deployed/maintained this type of system before?
```

Comment:
I don't know if a search index is something the platform should try to specialize in when there are viable third party options we can use. 

I suspect the bulk of the work is going to be cleaning up the input- indexing poorly structured, poorly edited content may result in a bad user experience. I assume that most of our documentation is in this state. 

I strongly recommend focusing on the part of this with the least amount of certainty (e.g. the indexing, the UX of the results) and to try to find a low effort way to validate or iterate on that. We should avoid committing any infrastructure to this until we have this working the way we expect in a local environment.